### ISSUE TITLE
PDF export: wrong colours

### ISSUE THREAD

**Issue by Hoffenbar** [giuspen/cherrytree#1311]
Hello, white text on black background is exported as black on black.....

**Comment by txe** [comment]
Hi, I don't think there is a general solution here.
I suppose, you have a dark theme with white text. Pdf format is used to print out documents on a white paper.  We don't want change background color of the whole pdf page just because the current theme is dark.

In your case, you just need to change not only the color of the text but its background too.

**Comment by Hoffenbar** [comment]
Hi,no, the general background is white. No dark theme!-- Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 08.11.20, 13:47 schrieb txe <notifications@github.com>:

   
  Hi, I don't think there is a general solution here. I suppose, you have a dark theme with white text. Pdf format is used to print out documents on a white paper. We don't want change background color of the whole pdf page just because the current theme is dark. 
  In your case, you just need to change not only the color of the text but its background too.

**Comment by txe** [comment]
@giuspen , it caused by 
```
Glib::ustring color_no_white = CtRgbUtil::rgb_to_no_white(property_value);
```
do you remember why to use this trick? maybe we can just remove it? HTML export has the same issue.

**Comment by giuspen** [comment]
@txe that is for the code syntax highlighting. When you have a dark background and white/light text/code, that won't be visible on the pdf/html white background. Maybe we can use only for the automatic syntax highlighting colours or we could improve to detect if the background is dark or not.

**Comment by txe** [comment]
are you sure? we take a color from tags, and syntax highlighting doesn't changes tags. 
I just removed that converting color code, and tried to export pdf: text color is default, syntax highlighting is white, as a result, the text in pdf is black as should be, not white.

**Comment by txe** [comment]
Hmm, maybe I'm missing something, codebox in pdf should have colors of highlighting

**Comment by giuspen** [comment]
Yes we are reading the colours straight from the gtk tags

**Comment by giuspen** [comment]
Only the special words have tags (like if, else, a string...) though, the default text no so we are not detecting that as white even if it is

**Comment by txe** [comment]
Well, just to make it clear, I was talking about Rich Text, but I think it applied to plain and other types of node.

If a theme changes text color and background, we still don't get it through our tags (tags in node textbuffer). So as long as user doesn't change them, they will be black and white in the code in spite of the theme. I just checked it.

Codebox has colorful output in pdf because we read `property_foreground_gtk` and use it to apply color to text in output. 

So, as I see, `rgb_to_no_white` can be removed because the text can be white if only user makes  it white.

**Comment by giuspen** [comment]
Ok

**Comment by Hoffenbar** [comment]
Hi,i don't meant a codebox. Just normal richtext, where a word could be configured to have white fg and black bg e.g.-- Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 08.11.20, 17:54 schrieb txe <notifications@github.com>:

   
  Hmm, maybe I'm missing something, codebox in pdf should have colors of highlighting

**Comment by txe** [comment]
Fixed. The fix will be in 0.99.20
